### DISCUSSION TITLE
Release v0.5.0

### DISCUSSION THREAD

**Discussion by chas-iot** [sqm-autorate/sqm-autorate#141] [Announcements]
If you run _sqm-autorate_ via `service sqm-autorate enable && service sqm-autorate start` then nothing changes, skip to **Changes**. If you use an alternate method and it fails, then see **Starting _sqm-autorate_**

# Changes
- relicense all code to MPLv2
- reduced the output to /tmp to prevent using up all available space and then crashing
  - reduced the amount of logging
  - new option to stop recording statistics
    - /etc/config/sqm-autorate, section output, option suppress_statistics, set to 1, y, yes or true
- many, many bugfixes
- updated the documentation in README.md
- command line settings
  - depends on the lua argparse module
  - `lua sqm-autorate.lua -h` for help
  - `lua sqm-autorate.lua -s` to see the internal settings
- environment variable settings
- streamlined the setup script
- started modularisation of the lua code

# Starting _sqm-autorate_
_sqm-autorate_ is now split into several modules. This requires some changes to the way the program is started so that `lua` can find and load the modules.
* If you only use `service sqm-autorate start` etc. then no change.
* If you start by typing `lua /usr/lib/sqm-autorate/sqm-autorate.lua`, then try `cd /usr/lib/sqm-autorate/ ; lua sqm-autorate`.
* If you start by typing `/usr/lib/sqm-autorate/sqm-autorate.lua` then type `cd /usr/lib/sqm-autorate/ ; ./sqm-autorate.lua`
* Alternatively, adjust the `LUA_PATH` to add `/usr/lib/sqm-autorate/?.lua`

# Commits
## Fail-Safe
- Fix for GH issue #99 per @jeverley
- Closes Missing curl command when running setup Fail-Safe/sqm-autorate#93
- Removed bad reflectors.
- Per @jeverley's feedback, there is no need for a pattern match here.
- This appears to only be needed for ingress. Removing egress reference.
- Fixed incorrectly named variable.
- Attempt at br-lan fix.

## Daniel Lakeland
- add some lines and text to bandwidth diagram
- add an example run bw graph for use in readme

## Nils Andreas Svee
- set names on threads for debugging
- Add more reflectors in APAC region

## jeverley
- Incorporate changes discussed in #112 in clean rebase to develop/main

## Rich Brown
- Main lua code: Add csv_logger() function that observes the desired log level (to minimize logging long-term)
  - note: later superseded
- Update getstats.sh

## Chas-IOT
- version v0.5.0
- fixup .gitattributes for graphics
- fix use of argparse and align documentation (#133)
- re-organise setup script - incorporated most changes by @richb-hanover
- Fix #131 - crash when environment variable not set (#132)
- documentation updates
- fix formatting
- automate enable_verbose_baseline_output
- quieten statistics with new setting - uci: output.suppress_statistics - cli: --suppress-statistics - env: SQMA_SUPPRESS_STATISTICS
- quieten logging
- make setup and service ready for modularisation
- improved language and layout in julia/datavis.html (#72)
- Modularisation stage 2 - settings (#123)
- Modularisation stage 1 - requires and utilities (#122)
- prevent pollution from windows dev environment (#129)
- relicense all code to MPLv2 add contact details for developers
- edit and re-organise the README (#113)
- Fix download locations for move to new organisation
- revision of the docs to align with settings

**Comment by malikshi** [comment]
Is it fine to install latest version?

**Comment by richb-hanover** [comment]
I have been running the 0.5.0 version branch for 24 hours. If you're brave, willing help "Science", able to report problems/errors, and willing to revert if you have serious problems, you should definitely go ahead and use the instructions at: 

https://github.com/sqm-autorate/sqm-autorate/tree/testing/lua-threads

**Reply by chas-iot** [reply]
I don’t believe that anyone needs to be brave to install this version.

While there are changes to the startup, once it’s running there’s no changes except bug fixes to the main algorithms.  So I’d definitely recommend it if you have installed previous versions or if you wish to explore this space.
